video/out/vulkan/context_mac.m

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "video/out/gpu/context.h"
+#include "osdep/macOS_swift.h"
+
+//#import <MoltenVK/mvk_vulkan.h>
+
+#include "common.h"
+#include "context.h"
+#include "utils.h"
+
+struct priv {
+    struct mpvk_ctx vk;
+    MacCommon *vo_macos;
+};
+
+
+static void macos_vk_uninit(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv;
+
+    ra_vk_ctx_uninit(ctx);
+    mpvk_uninit(&p->vk);
+    [p->vo_macos uninit:ctx->vo];
+}
+
+static bool macos_vk_init(struct ra_ctx *ctx)
+{
+    struct priv *p = ctx->priv = talloc_zero(ctx, struct priv);
+    struct mpvk_ctx *vk = &p->vk;
+    int msgl = ctx->opts.probing ? MSGL_V : MSGL_ERR;
+
+    if (!mpvk_init(vk, ctx, VK_MVK_MACOS_SURFACE_EXTENSION_NAME))
+        goto error;
+
+    p->vo_macos = [[MacCommon alloc] init:ctx->vo];
+    if (!p->vo_macos)
+        goto error;
+
+    VkMacOSSurfaceCreateInfoMVK macos_info = {
+        .sType = VK_STRUCTURE_TYPE_MACOS_SURFACE_CREATE_INFO_MVK,
+        .pNext = NULL,
+        .flags = 0,
+        .pView = p->vo_macos.video,
```

Comment:
segfaulted, until I have changed it to p->vo_macos.video.layer

Reply:
that's kinda interesting since the docs say something else https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkMacOSSurfaceCreateInfoMVK.html

on the other hand the moltenVK docs say something else and the layer is recommended https://github.com/KhronosGroup/MoltenVK/pull/258/commits/e546b838524dc9e8fae63a7d1aeb35c2f60d4ea0

Reply:
https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkMetalSurfaceCreateInfoEXT.html
this one should be used with the layer directly.

Reply:
VK_MVK_macos_surface is called obsolete here https://github.com/KhronosGroup/Vulkan-Tools/pull/220, so maybe it's better to just use VK_EXT_metal_surface

Reply:
interesting. i was planning to use the Metal Layer directly at some point anyway (i looked into metal surface ext beforehand), to reduce some redundancy and make it similar to cocoa-cb with its layer.

Reply:
does that mean that you're planning to add vulkan support to libmpv?

Reply:
no, i don't think i have enough time for that.